src/test/java/LoginTests.java

```diff
@@ -1,25 +1,117 @@
 
 //      Added ChromeOptions argument below to fix websocket error
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--remote-allow-origins=*");
+        // navigateToPage();
+        // Assert.assertEquals(driver.getCurrentUrl(), url);
+        driver.quit();
+    }
+
+    @Test
+    public void loginValidEmailPassword() throws InterruptedException {
 
-        WebDriver driver = new ChromeDriver(options);
-        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        //STep:1 Open Browser and navigate to Koel app;
+        // navigateToPage();
+        //Step:2 Enter email
+        provideEmail("kaflimeerim@gmail.com");
+        //Step:3 Enter password
+        providePassword("te$t$tudent");
+        //Step:4 Click on login button
+        loginToKoel();
+        //Thread.sleep(2000);
+        //Step:5 Assertion (expected vs actual)
+        //WebElement avatarIcon = driver.findElement(By.cssSelector("img[class='avatar']"));
+        //ExplicitWait
+         WebElement avatarIcon = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[class='avatar']")));
+        //FluentWait
+       // WebElement avatarIcon = fluentWait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[class='avatar']")));
+        Assert.assertTrue(avatarIcon.isDisplayed());
+        // driver.quit();
+
+    }
+
+    @Test(dataProvider = "invalidLoginData",dataProviderClass =  TestData.class)
+    public void loginWithNegativeData(String email, String password) {
+        //  navigateToPage();
+        provideEmail(email);
+        providePassword(password);
+        loginToKoel();
+       // WebElement avatarIcon = driver.findElement(By.cssSelector("img[class='avatar']"));
+        WebElement avatarIcon = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[class='avatar']")));
+        Assert.assertTrue(avatarIcon.isDisplayed());
+        // driver.quit();
+
+    }
+
+    //Login with Valid Email,Password using Page Object Model;
+    @Test
+     public void loginValidEmailValidPasswordTest() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(getDriver());
+        HomePage homePage = new HomePage(getDriver());
+
+        loginPage.provideEmail("kaflimeerim@gmail.com");
+        loginPage.providePassword("te$t$tudent");
+        loginPage.clickSubmit();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+
+     }
+    @Test
+    public void loginInValidEmailValidPasswordTest() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(getDriver());
+        HomePage homePage = new HomePage(getDriver());
+
+        loginPage.provideEmail("kaflimeerimgmail.com");
+        loginPage.providePassword("te$t$tudent");
+        loginPage.clickSubmit();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
```

Comment:
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.